src/components/CompareResults/ResultsMain.tsx

```diff
@@ -91,18 +96,106 @@ function ResultsMain() {
     }
   }
 
+  /********** Edit Results Title Section **********/
+  const [rawSearchParams, updateRawSearchParams] = useRawSearchParams();
+  const initialComparisonTitle = rawSearchParams.get('title') ?? 'Results';
+  const [comparisonTitleName, setComparisonTitleName] = useState(
+    initialComparisonTitle,
+  );
+  const [previousComparisonTitle, setPreviousComparisonTitle] =
+    useState('Results');
+  const [editComparisonTitleInputVisible, showEditComparisonTitle] =
+    useState(false);
+
+  const handleEditInputToggle = () => {
```

Comment:
It would be easier to have 2 different functions, because it's used for 2 different things:
* display when not displayed -> `onEditButtonClick` (for example)
* canceled -> `onEditCancel` (for example)

(when it's saved it's another function, which adds to the confusion IMO)

This would remove the if condition, make the code more linear and easier to follow.